### PR TITLE
[Assessor] Added request form

### PR DESCRIPTION
**Pull request by thomasbisignani**
**- Workflow :**
workflow

**- Step 1 :**
Assesseur - Step 1

**- Step 2 :**
Assesseur - Step 2

**- Step 3 :**
Assesseur - Step 3

### CODE REVIEW

File: src/Assessor/AssessorRequestCommand.php [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+     */
+    private $firstName;
+
+    /**
+     * @Assert\Length(
+     *     min=2,
+     *     max=50,
+     *     minMessage="assessor.birth_name.min_length",
+     *     maxMessage="assessor.birth_name.max_length",
+     * groups={"fill_personal_info"})
```
Comment (Remg) [inline src/Assessor/AssessorRequestCommand.php]: missing indentation

File: src/Assessor/AssessorRequestCommand.php [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+     *     callback={"AppBundle\Entity\AssessorOfficeEnum", "toArray"},
+     *     message="assessor.office.invalid_choice",
+     *     strict=true,
+     *     groups={"fill_assessor_info"}
+     * )
+     */
+    private $office = AssessorOfficeEnum::HOLDER;
+
+    /**
+     * @Assert\NotBlank(groups={"fill_assessor_info"})
```
Comment (Remg) [inline src/Assessor/AssessorRequestCommand.php]: no custom message for this one ?

File: src/Assessor/AssessorRequestFactory.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            $data['phoneNumber'],
+            $data['assessorCity'],
+            $data['assessorPostalCode'],
+            $data['office']
+        );
+
+        return $assessor;
+    }
+
+    public function createFromCommand(AssessorRequestCommand $assessorRequestCommand): AssessorRequest
```
Comment (Remg) [inline src/Assessor/AssessorRequestFactory.php]: why do you use setters here, and constructor on the method above ?

File: src/Assessor/AssessorRequestHandler.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        VotePlaceManager $votePlaceManager,
+        AssessorRequestFactory $assessorRequestFactory
+    ) {
+        $this->stateMachine = $stateMachine;
+        $this->manager = $manager;
+        $this->votePlaceManager = $votePlaceManager;
+        $this->assessorRequestFactory = $assessorRequestFactory;
+    }
+
+    public function handle(SessionInterface $session, AssessorRequestCommand $assessorRequestCommand): bool
```
Comment (ottaviano) [inline src/Assessor/AssessorRequestHandler.php]: why the session is not a class dependency?

File: src/VotePlace/VotePlaceManager.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    private $repository;
+
+    public function __construct(VotePlaceRepository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function getVotePlaceWishesLabelsByIds(array $ids): array
+    {
+        return array_map(function (VotePlace $votePlace) {
```
Comment (ottaviano) [inline src/VotePlace/VotePlaceManager.php]: this method executes a query, hydrates a Place objects for keeping only name value
what about `queryBuilder->select(name)->getArrayResult()`

File: templates/assessor/_fill_assessor_info.html.twig [hunk 6]
```diff
@@ -16,15 +16,15 @@
             </div>
         </div>
 
-        <div class="l__row l__row--top l__row--h-stretch l__mobile--col">
+        <div class="french-bloc l__row l__row--top l__row--h-stretch l__mobile--col">
```
Comment (ottaviano) [inline templates/assessor/_fill_assessor_info.html.twig]: block

File: templates/assessor/_fill_assessor_info.html.twig [hunk 7]
```diff
@@ -16,15 +16,15 @@
...
             <div class="l__row--h-stretch register__form--full">
                 <p class="b__nudge--bottom-tiny text--bold">Code postal</p>
                 {{ form_widget(form.assessorPostalCode, { attr: { placeholder: 'Code postal où vous souhaitez être assesseur(e)' } }) }}
                 {{ form_errors(form.assessorPostalCode) }}
             </div>
         </div>
 
-        <div class="l__row l__row--top l__row--h-stretch l__mobile--col">
+        <div class="french-bloc l__row l__row--top l__row--h-stretch l__mobile--col">
```
Comment (ottaviano) [inline templates/assessor/_fill_assessor_info.html.twig]: block

File: templates/assessor/request.html.twig [hunk 8]
```diff
@@ -18,31 +18,60 @@
...
-                if ($(this).val().length > 4) {
+            const assessorCountrySelect = $('#assessor_request_assessorCountry');
+            const assessorPostalCodeInput = $('#assessor_request_assessorPostalCode');
+            const assessorCityInput = $('#assessor_request_assessorCity');
 
-                    let url = "{{ path('app_assessor_request_find_vote_places', { postalCode: 'postalCodeVal' }) }}";
+            var initForeignAssessorRequest = function () {
+                assessorPostalCodeInput.val('');
+                assessorCityInput.val('');
+                $('.french-bloc').hide();
```
Comment (ottaviano) [inline templates/assessor/request.html.twig]: block

File: templates/assessor/request.html.twig [hunk 9]
```diff
@@ -18,31 +18,60 @@
...
-                        $.each(data, function (value, key) {
-                            votePlaces.push({
-                                id: key,
-                                text: value,
-                            });
-                        });
+            if (assessorCountrySelect.val() !== 'FR') {
+                initForeignAssessorRequest();
+            } else {
+                $('.french-bloc').show();
```
Comment (ottaviano) [inline templates/assessor/request.html.twig]: same

**Comment by thomasbisignani** [comment]
Merged in https://github.com/EnMarche/en-marche.fr/pull/3571

**Pull request closed without merging**